spinalcordtoolbox/deepseg/inference.py

```diff
@@ -83,3 +94,171 @@ def segment_and_average_volumes(model_paths, input_filenames, options):
     im_lst = [Image(np.asanyarray(nii.dataobj), hdr=nii.header) for nii in nii_lst]
 
     return im_lst, target_lst
+
+
+def segment_non_ivadomed(path_model, model_type, input_filenames, threshold):
+    # MONAI and NNUnet have similar structure, and so we use nnunet+inference functions with the same signature
+    if model_type == "monai":
+        create_net = ds_monai.create_nnunet_from_plans
+        inference = segment_monai
+    else:
+        assert model_type == "nnunet"
+        create_net = ds_nnunet.create_nnunet_from_plans
+        inference = segment_nnunet
+
+    # equivalent to `with torch.no_grad()`
+    torch.set_grad_enabled(False)
+
+    # load model from checkpoint
+    net = create_net(path_model)
+
+    im_lst, target_lst = [], []
+    for fname_in in input_filenames:
+        # model may be multiclass, so the `inference` func should output a list of fnames and targets
+        fnames_out, targets = inference(path_img=fname_in, tmpdir=tmp_create(basename="sct_deepseg"), predictor=net)
```

Comment:
Should this temp directory get cleaned up at some point?

Reply:
You know, that's a very good point! I always assumed that tmpdirs were automatically cleaned up by the OS, so I usually don't touch them (e.g. when using `tmp_path` in tests).

But, in scripts, we have the `-r` option for a reason:

```
  -r {0,1}              Remove temporary files. (default: 1)
```

So, we probably should be using this to manage directories created by `tmp_create`! (I think I've largely forgotten to use this option, to be quite honest.)